app/controllers/BillsController.scala

```diff
@@ -41,15 +44,22 @@ class BillsController @Inject()(implicit val config: FrontendAppConfig,
                                ) extends BaseController {
 
   val viewCrystallisedCalculations: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino andThen retrieveIncomeSources).async {
-    implicit user =>
-      implicit val sources: IncomeSourcesModel = user.incomeSources
-
-      for {
-        lastTaxCalcs <- calculationService.getAllLatestCalculations(user.nino, sources.orderedTaxYears)
-      } yield {
-          Logger.debug(s"[BillsController][viewCrystallisedCalculations] Retrieved Last Tax Calcs With Year response: $lastTaxCalcs")
-          if (lastTaxCalcs.exists(_.isErrored)) itvcErrorHandler.showInternalServerError
-          else Ok(views.html.bills(lastTaxCalcs.filter(_.matchesStatus(Crystallised))))
-      }
+    implicit user => if(config.features.billsEnabled()) renderView else redirectToHome
+  }
+
+  private[BillsController] def redirectToHome: Future[Result] = {
+    Future.successful(Redirect(controllers.routes.HomeController.home().url))
+  }
+
+  private[BillsController] def renderView[A](implicit user: MtdItUser[A]): Future[Result] = {
+    implicit val sources: IncomeSourcesModel = user.incomeSources
+
+    for {
```

Comment:
As there is only one future within this `for` comprehension it feels like overkill to use this.

Instead, you could just use `.map` on the call to the service. I.e.:
```
    calculationService.getAllLatestCalculations(user.nino, sources.orderedTaxYears).map {
      case lastTaxCalcs if lastTaxCalcs.exists(_.isErrored) =>
        Logger.debug(s"[BillsController][viewCrystallisedCalculations] Retrieved at least one Errored Last Tax Calc. Response: $lastTaxCalcs")
        itvcErrorHandler.showInternalServerError
      case lastTaxCalcs =>
        Ok(views.html.bills(lastTaxCalcs.filter(_.matchesStatus(Crystallised))))
    }
```